api/v1alpha1/ingressnodefirewall_webhook.go

```diff
@@ -43,22 +53,175 @@ var _ webhook.Validator = &IngressNodeFirewall{}
 func (r *IngressNodeFirewall) ValidateCreate() error {
 	ingressnodefirewalllog.Info("validate create", "name", r.Name)
 
-	// TODO(user): fill in your validation logic upon object creation.
-	return nil
+	return validateIngressNodeFirewall(r)
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *IngressNodeFirewall) ValidateUpdate(old runtime.Object) error {
 	ingressnodefirewalllog.Info("validate update", "name", r.Name)
 
-	// TODO(user): fill in your validation logic upon object update.
-	return nil
+	return validateIngressNodeFirewall(r)
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type
 func (r *IngressNodeFirewall) ValidateDelete() error {
 	ingressnodefirewalllog.Info("validate delete", "name", r.Name)
 
-	// TODO(user): fill in your validation logic upon object deletion.
 	return nil
 }
+
+func validateIngressNodeFirewall(inf *IngressNodeFirewall) error {
+	if allErrs := validateINFRules(inf.Spec.Ingress, inf.Name); len(allErrs) > 0 {
+		return apierrors.NewInvalid(
+			schema.GroupKind{Group: GroupVersion.Group, Kind: IngressNodeFirewall{}.Kind},
+			inf.Name, allErrs)
+	}
+	return nil
+}
+
+func validateINFRules(infRules []IngressNodeFirewallRules, infName string) field.ErrorList {
+	var allErrs field.ErrorList
+	for infRulesIndex, infRule := range infRules {
+		if newErrs := validateFromCIDRS(allErrs, infRule.FromCIDRs, infRulesIndex, infName); len(newErrs) > 0 {
```

Comment:
I have been asked to chaneg fromCIDR to sourceCIDR so probably need to align the function name as well

Reply:
Will rebase once the PR is merged.

Reply:
Done